cmd/add.go

```diff
@@ -42,13 +55,75 @@ func runAdd(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	err = lock.AddResource(args, algo, tags, filename)
+	cache, err := cmd.Flags().GetString("cache")
 	if err != nil {
 		return err
 	}
+
+	// Check if ARTIFACTORY_TOKEN is set if cache URL is provided
+	token := os.Getenv("ARTIFACTORY_TOKEN")
+	if token == "" && cache != "" {
+		log.Warn().Msg("ARTIFACTORY_TOKEN is not set; cache upload may fail if authentication is required.")
```

Comment:
This should be an error, right?  I don't think there will ever be a case where you can upload to Artifactory without authentication.